frontend/packages/dev-console/src/components/monitoring/alerts/monitoring-alerts-utils.tsx

```diff
@@ -145,11 +151,41 @@ export const monitoringAlertRows = (
   return rows;
 };
 
+export const getFilteredRows = (_filters, rowFilters, objects) => {
+  if (_.isEmpty(_filters)) {
+    return objects;
+  }
+
+  const allTableFilters = getAllTableFilters(rowFilters);
+  let filteredObjects = objects;
+  _.each(_filters, (value, name) => {
+    const filter = allTableFilters[name];
+    if (_.isFunction(filter)) {
+      if (name === 'alert-state') {
+        filteredObjects = _.filter(filteredObjects, (o) =>
+          !_.isEmpty(value.selected)
+            ? _.filter(o.alerts, (a) => filter(value, a)).length > 0 ||
+              (value.selected.has('inactive') && _.isEmpty(o.alerts))
+            : o,
+        );
```

Comment:
Maybe its just for myself to understand this.. If none filter is selected, you use the o (entry in filteredObjects) as filter value. Isn't it easier and cleaner to wrap this in an if, in the case we want keep 'all' filteredObjects? WDYT? :thinking: 

```jsx
    if (_.isFunction(filter)) {
      if (name === 'alert-state') {
        if (_.isEmpty(value.selected)) {
          filteredObjects = _.filter(filteredObjects, (o) =>
            _.filter(o.alerts, (a) => filter(value, a)).length > 0 ||
              (value.selected.has('inactive') && _.isEmpty(o.alerts)),
          );
        }
      } else {
        filteredObjects = _.filter(filteredObjects, (o) => filter(value, o));
      }
    }
```

Reply:
Agree, Thanks for the suggestion.